### PR TITLE
Make validation a pass and add it after every pass by default

### PR DESCRIPTION
**Pull request by rkimballn1**
This is an alternative way to do validation between passes. If you don't like it we can close the PR.

### CODE REVIEW

File: src/ngraph/pass/manager.hpp [hunk 1]
```diff
@@ -62,20 +61,30 @@ class ngraph::pass::Manager
             m_pass_names.push_back(typeid(T).name());
 #endif
         }
+        if (m_per_pass_validation && !std::is_same<T, Validate>::value)
```
Comment (aprocter) [inline src/ngraph/pass/manager.hpp]: It's not a big deal, but I wonder if this is overkill.

File: src/ngraph/pass/validate.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    namespace pass
+    {
+        class Validate : public FunctionPass
+        {
+        public:
+            Validate()
+                : FunctionPass()
+            {
+            }
+            virtual bool run_on_function(std::shared_ptr<ngraph::Function> f) override;
```
Comment (earhart) [inline src/ngraph/pass/validate.hpp]: `virtual override`?  (Why not just `final`?)

File: src/ngraph/pass/manager.hpp (hunk 1 above)
Comment (rkimballn1) [reply src/ngraph/pass/manager.hpp]: I was wondering the same thing too. I doubt that users will use the "auto insert validation" mode along with adding Validate passes into the graph. I will remove the `is_same`. Double validation never killed anyone.

File: src/ngraph/pass/validate.hpp (hunk 2 above)
Comment (rkimballn1) [reply src/ngraph/pass/validate.hpp]: Ugh. That's what I get for copy/paste. I will remove virtual.

File: src/ngraph/pass/manager.cpp [hunk 3]
```diff
@@ -53,11 +53,7 @@ pass::Manager::~Manager()
...
 }
 
-void pass::Manager::initialize_default_passes()
-{
-}
-
-void pass::Manager::run_passes(shared_ptr<Function> func, bool transitive, bool revalidate)
+void pass::Manager::run_passes(shared_ptr<Function> func, bool transitive)
 {
     bool profile_enabled = getenv("NGRAPH_PROFILE_PASS_ENABLE") != nullptr;
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/pass/manager.cpp]: done

**Pull request merged**